### PR TITLE
Fix amp-brightcove unlayoutCallback and multiple layout calls

### PR DESCRIPTION
**Pull request by mkhatib**
Putting this fix in a separate PR from #2753

### CODE REVIEW

File: extensions/amp-brightcove/0.1/amp-brightcove.js [hunk 1]
```diff
@@ -34,7 +34,19 @@ class AmpBrightcove extends AMP.BaseElement {
...
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+  }
+
+  /** @override */
   layoutCallback() {
+    if (this.iframePromise_) {
```
Comment (dvoytenko) [inline extensions/amp-brightcove/0.1/amp-brightcove.js]: Why is this? `layoutCallback` is only allowed to be called once for this element.

Comment (dvoytenko) [inline extensions/amp-brightcove/0.1/amp-brightcove.js]: I see. You are protecting against re-layout after unlayout which is allowed. However, this code is problematic - it actually prevents relayout-after-layout from happening.

Comment (mkhatib) [inline extensions/amp-brightcove/0.1/amp-brightcove.js]: hmm actually I thought layout will be called multiple times. I think I got this impression from the tests having created multiple iframes - checking again it seems the test actually calls layoutCallback again manually without turning off runtime. Will fix that instead.

Comment (mkhatib) [inline extensions/amp-brightcove/0.1/amp-brightcove.js]: Done.

**Comment by mkhatib** [comment]
PTAL

**Comment by dvoytenko** [comment]
LGTM

**Comment by mkhatib** [comment]
Merged prematurely - didn't squash.

**Pull request merged**